### ISSUE TITLE
Get-AzAdApplication -OwnedApplication returns an array instead of the array elements

### ISSUE THREAD

**Issue by Alex-wdy** [Azure/azure-powershell#23644]
### Description

I have 46 owned applications, but instead of getting them, I get a single array that contains those 46 applications.

Get-AzADApplication | gm

   TypeName: Microsoft.Azure.PowerShell.Cmdlets.Resources.MSGraph.Models.ApiV10.MicrosoftGraphApplication

Get-AzADApplication -OwnedApplication | gm

   TypeName: System.Object[]

Adding the -OwnedApplication switch should not change the output type.

![WeChat Image_20231204080731](https://github.com/Azure/azure-powershell/assets/108926104/b5e533b7-118a-4047-991c-de76bc84e190)

### Issue script & Debug output

```PowerShell
Get-AzADApplication | gm

   TypeName: Microsoft.Azure.PowerShell.Cmdlets.Resources.MSGraph.Models.ApiV10.MicrosoftGraphApplication

Get-AzADApplication -OwnedApplication | gm

   TypeName: System.Object[]

Adding the -OwnedApplication switch should not change the output type.
```

### Environment data

```PowerShell
PS7.3
```

### Module versions

```PowerShell
Az.Resource
```

**Comment by isra-fel** [comment]
This looks like a bug. Adding a filter should not change the return type. We'll look into it.

**Comment by dolauli** [comment]
Fix will be in the coming release Az 11.2.0.
